chore(dracut-init.sh): remove support for obsolete syntax

LaszloGombos opened the pull request:
Old syntax dracut modules had its check install installkernel scripts in dedicated shell files and did not support module dependencies.

For fun - here is the commit that transitioned from obsolete syntax to the current syntax from [2011](https://github.com/dracutdevs/dracut/commit/95d2dabc250b39d2452e0e8f57a702ab80db895c)

This type of dracut modules are not even documented any more in the documentation - https://github.com/dracutdevs/dracut/blob/master/man/dracut.modules.7.asc

## Checklist
- [X] I have tested it locally
- [X] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

Fixes #2186

aafeijoo-suse reviewed (commented):
If we decide to remove this, I think we should `exit` dracut if `! -f $_moddir/module-setup.sh`.

aafeijoo-suse reviewed (commented):
> If we decide to remove this, I think we should exit dracut if ! -f $_moddir/module-setup.sh.

Ok, I guess a `return 1` is enough.

You can also fix these lines:

https://github.com/dracutdevs/dracut/blob/e9b4774239989257999053033fc92cded7803896/dracut-init.sh#L1073

https://github.com/dracutdevs/dracut/blob/e9b4774239989257999053033fc92cded7803896/dracut.sh#L1380

LaszloGombos committed:
chore(dracut-init.sh): remove support for obsolete syntax

Old syntax dracut modules had its check install installkernel
scripts in dedicated shell files and did not support module
dependencies.
